docs/DeveloperGuide.md

```diff
@@ -1,13 +1,173 @@
+You should see the following greeting message if the project setup is successful:
+
+```
+  __  __                         _____   __          ___                   
+ |  \/  |                       / ____|  \ \        / / |                  
+ | \  / | ___  _ __   ___ _   _| |  __  __\ \  /\  / /| |__   ___ _ __ ___ 
+ | |\/| |/ _ \| '_ \ / _ \ | | | | |_ |/ _ \ \/  \/ / | '_ \ / _ \ '__/ _ \
+ | |  | | (_) | | | |  __/ |_| | |__| | (_) \  /\  /  | | | |  __/ | |  __/
+ |_|  |_|\___/|_| |_|\___|\__, |\_____|\___/ \/  \/   |_| |_|\___|_|  \___|
+                           __/ |                                           
+                          |___/                                            
+
+Your MoneyGoWhere? Let me help you track it.
+```
+
+## Design
+### Software Architecture:
+The software architecture diagram below describes the program's design and the interaction between components.
+
+![Software-Architecture](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/SoftwareArchitecture.puml)
+
+### Core Components:
+
+| Component     | Function                                                                            |
+|---------------|-------------------------------------------------------------------------------------|
+| MoneyGoWhere  | Main entrypoint of the program.                                                     |
+| Common        | Defines various parameters used by the program.                                     |
+| Exceptions    | Defines exceptions thrown by the program.                                           |
+| UserInterface | Provides functions to interface with the user via standard I/O and handle commands. |
+| Command       | Defines the commands accepted by the program along with its arguments.              |
+| Parser        | Provides functions to parse inputs read from standard input.                        |
+| Data          | Stores data and provides functions to operate on data.                              |
+| Storage       | Defines functions to save and load data.                                            |
+| Logger        | Defines functions to log the user's actions and the program's behaviour.            |
+
+### Component Interactions:
+The sequence diagram below describes the interaction between the various core components when a command is entered.
+In this example, 
+the user launches the program and enters the command `Add-Expense -n Expense -a 7.80` to add an expense with the name `Expense` and the amount `7.80`.
+The sequence diagrams referenced by the component interaction diagram can be seen [below](#component-interaction-sequence-diagrams)
+
+![Component-Interaction-On-Command-Entered](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/xzynos/tp/branch-MoneyGoWhere-Webpage/docs/diagrams/ComponentInteractionsOnCommandEntered.puml)
```

Comment:
![image](https://user-images.githubusercontent.com/21076314/198246467-d806093c-7afc-45d4-a680-bd13dc850401.png)
Should this instance of MoneyGoWhere be created by the user instead?